README.md

```diff
@@ -1,62 +1,356 @@
+    ISPublisher* create_publisher(ISBridge* bridge, const char* name,
+    	std::vector<std::pair<std::string, std::string>> *configuration);
+
+The function *create_bridge* must return a pointer to an instance of a derived class of ISBridge, or nullptr if failed.
+The function *create_subscriber* must return a pointer to an instance of a derived class of ISSubscriber, or nullptr if failed.
+The function *create_publisher* must return a pointer to an instance of a derived class of ISPublisher, or nullptr if failed.
+Integration Services will deallocate this objects from memory when the bridge is stopped.
+
+Let's take a look to this interfaces:
+
+	/** Base class for publishers. Must know how to write into the destination protocol */
+	class ISPublisher
+	{
+	public:
+	    ISPublisher(const std::string &name);
+	    virtual ~ISPublisher() = default;
+	    virtual bool publish(SerializedPayload_t *data);
+	    virtual ISBridge* setBridge(ISBridge *bridge);
+	};
+
+	/** Base class for subscribers. Must know how to read from the origin protocol */
+	class ISSubscriber
+	{
+	public:
+	    ISSubscriber(const std::string &name);
+	    virtual ~ISSubscriber() = default;
+	    virtual void addBridge(ISBridge* bridge);
+	    virtual void on_received_data(SerializedPayload_t* payload);
+	};
+
+	/**
+	 * Base class for Bridges. All implementation must inherit from it.
+	 */
+	class ISBridge
+	{
+	public:
+	    ISBridge(const std::string &name);
+	    virtual ~ISBridge();
+	    virtual void addSubscriber(ISSubscriber *sub);
+	    virtual void addFunction(const std::string &sub, const std::string &fname, userf_t func);
+	    virtual void addPublisher(const std::string &sub, const std::string &funcName, ISPublisher* pub);
+	    virtual ISPublisher* removePublisher(ISPublisher* pub);
+	    virtual void on_received_data(const ISSubscriber *sub, SerializedPayload_t *data);
+	    virtual void onTerminate() override;
+	};
+
+When implementing your ISBridge derived class, you must take in account:
+
+- Only ISPublisher::publish is mandatory to implement.
+- When your subscriber receives data, you must call on_received_data function with the data properly converted into SerializedPayload_t.
+- You can override the default behaviour, but isn't recommended in general. This behaviour follows this diagram:
+
+```plantuml
+@startuml
+hide footbox
+
+participant ISSubscriber
+participant ISBridge
+participant TransformationLibrary
+participant ISPublisher
+
+[-> ISSubscriber : receive
+
+ISSubscriber -> ISSubscriber : on_received_data
+ISSubscriber -> ISBridge : on_received_data
+ISBridge -> TransformationLibrary : transform
+TransformationLibrary --> ISBridge : <<transform_return>>
+ISBridge -> ISPublisher : publish
+
+ISPublisher ->] : write
+@enduml
+```
+
+When the subscriber calls it self to *on_received_data*, it will call all the *bridges* it belongs, calling the method *on_received_data* of each bridge.
```

Comment:
```suggestion
When the subscriber calls to its method *on_received_data*, it will call all the *bridges* it belongs, calling the method *on_received_data* of each bridge.
```